static/src/javascripts/projects/common/modules/navigation/user-account.js

```diff
@@ -2,110 +2,53 @@
-
-    const preloadAvatar = (src: string): Promise<void> => {
-        const image = new Image();
-        image.src = src;
-
-        return new Promise(resolve => {
-            image.onload = resolve;
-        });
-    };
+    const commentItems = [
+        ...document.querySelectorAll('.js-show-comment-activity'),
+    ];
+    const user = getUserFromCookie();
 
-    const swapFallback = (
-        fallback: ?HTMLElement,
-        avatar: ?HTMLElement,
-        src: string
-    ) => {
-        fastdom.write(() => {
-            if (fallback) {
-                fallback.classList.add('u-h');
-            }
+    if (user) {
+        commentItems.forEach(commentItem => {
+            const commentLink = commentItem.querySelector(
+                '.js-add-comment-activity-link'
+            );
 
-            if (avatar) {
-                avatar.setAttribute('src', src);
-                avatar.classList.remove('u-h');
+            if (commentLink) {
+                fastdom.write(() => {
+                    commentItem.classList.remove('u-h');
+                    commentLink.setAttribute(
+                        'href',
+                        `https://profile.theguardian.com/user/id/${user.id}`
+                    );
+                });
             }
         });
-    };
-
-    if (!isUserLoggedIn() || !avatarEl) {
-        return Promise.resolve();
     }
-
-    return avatarAPI.getActive().then(res => {
-        const src = res && res.data && res.data.avatarUrl;
-
-        if (src) {
-            preloadAvatar(src).then(() =>
-                swapFallback(fallbackEl, avatarEl, src)
-            );
-        }
-    });
 };
 
 const showMyAccountIfNecessary = (): void => {
     if (!isUserLoggedIn()) {
         return;
     }
 
-    const signIn = document.querySelector('.js-navigation-sign-in');
-    const accountDetails = document.querySelector(
-        '.js-navigation-account-details'
-    );
-    const accountActions = document.querySelector(
-        '.js-navigation-account-actions'
-    );
-    const user = getUserFromCookie();
-    const userNameEl = document.querySelector(
-        '.js-navigation-account-username'
-    );
+    const signIns = [...document.querySelectorAll('.js-navigation-sign-in')];
+    const accountActionsLists = [
+        ...document.querySelectorAll('.js-navigation-account-actions'),
+    ];
 
     fastdom.write(() => {
-        if (signIn) {
+        signIns.forEach(signIn => {
             signIn.classList.add('u-h');
-        }
-
-        if (accountDetails) {
-            accountDetails.classList.remove('u-h');
-        }
+        });
 
-        if (accountActions) {
+        accountActionsLists.forEach(accountActions => {
             accountActions.classList.remove('u-h');
-            updateCommentLink();
-        }
+        });
 
-        if (userNameEl && user && user.displayName) {
-            userNameEl.innerText = user.displayName;
-        }
+        updateCommentLink();
```

Comment:
Maybe we could move this out of `fastdom.write(...)` and into a chained `.then(...)` callback. Then we can use `fastdom.read(...)` for DOM reads inside `updateCommentLink()`